File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/writing_data/sample2.py
import dv_processing as dv

capture = dv.io.CameraCapture()

writer = dv.io.MonoCameraWriter("mono_writer_sample.aedat4", capture)

print(f"Is event stream available? {str(writer.isEventStreamConfigured())}")
print(f"Is frame stream available? {str(writer.isFrameStreamConfigured())}")
print(f"Is imu stream available? {str(writer.isImuStreamConfigured())}")
print(f"Is trigger stream available? {str(writer.isTriggerStreamConfigured())}")

File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/writing_data/sample2.cpp
#include <dv-processing/io/camera_capture.hpp>
#include <dv-processing/io/mono_camera_writer.hpp>

int main() {
	// Open any camera
	dv::io::CameraCapture capture;

	// Create the writer instance, writer will inspect the capture capabilities and create output
	// streams for all available data streams from the capture instance.
	dv::io::MonoCameraWriter writer("mono_writer_sample.aedat4", capture);

	// Print which streams were configured
	std::cout << std::boolalpha;
	std::cout << "Is event stream available? " << writer.isEventStreamConfigured() << std::endl;
	std::cout << "Is frame stream available? " << writer.isFrameStreamConfigured() << std::endl;
	std::cout << "Is imu stream available? " << writer.isImuStreamConfigured() << std::endl;
	std::cout << "Is trigger stream available? " << writer.isTriggerStreamConfigured() << std::endl;

	return 0;
}
